File: generated-code/python/codecraft/TcpReadWrite/vec2_float.py
from stream_wrapper import StreamWrapper

class Vec2Float:
    """2 dimensional vector."""

    __slots__ = ("x","y",)

    x: float
    y: float

    def __init__(self, x: float, y: float):
        self.x = x
        """`x` coordinate of the vector"""
        self.y = y
        """`y` coordinate of the vector"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "Vec2Float":
        """Read Vec2Float from input stream
        """
        x = stream.read_float()
        y = stream.read_float()
        return Vec2Float(x, y)
    
    def write_to(self, stream: StreamWrapper):
        """Write Vec2Float to output stream
        """
        stream.write_float(self.x)
        stream.write_float(self.y)
    
    def __repr__(self):
        return "Vec2Float(" + \
            repr(self.x) + \
            ", " + \
            repr(self.y) + \
            ")"

File: generated-code/php/codecraft/TcpReadWrite/Vec2Float.php
<?php

namespace  {
    

    /**
     * 2 dimensional vector.
     */
    class Vec2Float
    {
        /**
         * `x` coordinate of the vector
         */
        public $x;
        /**
         * `y` coordinate of the vector
         */
        public $y;
    
        function __construct($x, $y)
        {
            $this->x = $x;
            $this->y = $y;
        }
    
        /**
         * Read Vec2Float from input stream
         */
        public static function readFrom($stream)
        {
            $x = $stream->readFloat32();
            $y = $stream->readFloat32();
            return new Vec2Float($x, $y);
        }
    
        /**
         * Write Vec2Float to output stream
         */
        public function writeTo($stream)
        {
            $stream->writeFloat32($this->x);
            $stream->writeFloat32($this->y);
        }
    }
}